replica/table.cc

```diff
@@ -2044,20 +2044,42 @@ future<db::replay_position> table::discard_sstables(db_clock::time_point truncat
             };
             prune(st.main);
             prune(st.maintenance);
+        }
+    };
 
-            cg.set_main_sstables(std::move(st.main.pruned));
-            cg.set_maintenance_sstables(std::move(st.maintenance.pruned));
+    struct updater_impl : public row_cache::external_updater_impl {
+        table& t;
+        pruner& p;
+        db_clock::time_point truncated_at;
+        std::unordered_map<compaction_group*, compaction_backlog_tracker> new_trackers;
+
```

Comment:
it seems you forgot to update backlog trackers here instead of post cache update when deleting the dropped sstables